adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -14,224 +14,143 @@
   if (state == kUnityAdsFinishStateCompleted) {
     [_adEventDelegate didEndVideo];
-
+    
     // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
     // amount of 1. Publishers using this adapter should override the reward on the AdMob
     // front-end.
     GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
                                                      rewardAmount:[NSDecimalNumber one]];
     [_adEventDelegate didRewardUserWithReward:reward];
   } else if (state == kUnityAdsFinishStateError) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorFinish,
-        @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
+    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(GADMAdapterUnityErrorFinish,
+                                                                 @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
     [_adEventDelegate didFailToPresentWithError:error];
   }
-
+  
   [_adEventDelegate willDismissFullScreenView];
   [_adEventDelegate didDismissFullScreenView];
 }
 
 - (void)unityAdsDidStart:(nonnull NSString *)placementID {
-  if ([placementID isEqualToString:_placementID]) {
-    [_adEventDelegate didStartVideo];
-  }
+  [_adEventDelegate didStartVideo];
 }
 
 - (void)unityAdsReady:(nonnull NSString *)placementID {
-  if (_adLoadCompletionHandler && [placementID isEqualToString:_placementID]) {
-    _adEventDelegate = _adLoadCompletionHandler(self, nil);
-  }
 }
 
 - (void)unityAdsDidClick:(nonnull NSString *)placementID {
   // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
   // that a click event indicates the user is leaving the application for a browser or deeplink, and
   // notifies the Google Mobile Ads SDK accordingly.
-  if ([placementID isEqualToString:_placementID]) {
-    [_adEventDelegate reportClick];
-  }
+  [_adEventDelegate reportClick];
 }
 
 - (void)unityAdsPlacementStateChanged:(nonnull NSString *)placementID
                              oldState:(UnityAdsPlacementState)oldState
                              newState:(UnityAdsPlacementState)newState {
-  if (![placementID isEqualToString:_placementID]) {
-    return;
-  }
-  if (newState == kUnityAdsPlacementStateNoFill) {
-    if (_adLoadCompletionHandler) {
-      NSString *errorMsg =
-          [NSString stringWithFormat:@"No ad available for this placement ID: %@", placementID];
-      NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-          GADMAdapterUnityErrorPlacementStateNoFill, errorMsg);
-      _adLoadCompletionHandler(nil, error);
-    }
-    [[GADMAdapterUnitySingleton sharedInstance] stopTrackingDelegate:self];
-    return;
+}
+
+- (void)unityAdsAdFailedToLoad:(nonnull NSString *)placementId { 
+  if (_adLoadCompletionHandler) {
+    NSError *error = GADUnityErrorWithDescription(@"unityAdsAdFailedToLoad");
```

Comment:
Does Unity SDK provide a way to know the reason for the underlying error? In any case, please update the error message. Maybe something like `Failed to load rewarded ad`.

Reply:
It currently does not provide a reason for the underlying error.  But I will update the description here to be more friendly and include the placement ID.